ls1intum/Artemis
# `Exam mode`: Don't show exams to student if not registered

### Checklist
#### General


#### Server

### Motivation and Context


#6054

### Description

This PR implements the side for students of the linked issue. I.E. only exams are shown where the student is registered.

### Steps for Testing

Prerequisites:
- 1 Instructor
- 1 Student

1. Log in to Artemis
2. Create two exams, one where the student is registered and another one where he is not
3. Make sure that the visibility date for both exams is set in the past
4. Check that the student can only see the exam in the exams tab where he is registered

### Review Progress






#### Code Review
#### Manual Tests

### Test Coverage


| Class/File | Branch | Line |
|------------|-------:|-----:|
| CourseService.java | 78% | 93% |
| CourseResource.java | 74% | 94% |
